zvariant/src/signature.rs

```diff
@@ -288,6 +293,18 @@ impl<'a> Signature<'a> {
 
         clone
     }
+
+    /// Returns the number of complete types for the signature.
+    ///
+    /// **Note**: if the signature is invalid, returns the first error.
+    pub fn n_types(&self) -> Result<usize> {
```

Comment:
I think `n_signatures` would be more appropriate.

Reply:
the spec is not thinking about the context. Here, reader would think you're talking about Rust types.

Reply:
No.. it's a dbus signature. Nothing to do with Rust. If you want we refer to the spec from here.

Reply:
ok with the current version? would you rename it n_complete_types() instead?

Reply:
`n_complete_types` definitely seems better IMO.

Reply:
done